### PR TITLE
Various editor and UI improvements

### LINKED ISSUE #32: Improve editor validation
Currently, the event editor does no local validation. This should be improved, likely by using [Yup](https://github.com/jquense/yup) or something similar. This will, unfortunately, cause some code duplication between model definitions and input validation.

It may make sense to implement the validation into the API package after doing #28, as then we could also provide useful error messages from the API without even more code duplication or cross-dependencies.

### LINKED ISSUE #64: Add option to preview events as admin
Admins should be able to view events outside the editor, including draft and past events. This can use the current SingleEvent view with minimal modifications.

### PR DESCRIPTION
**Pull request by PurkkaKoodari**
This PR fixes many current annoyances and adds some features we can add without changing the (effective) DB or API schema. (There are some API changes, but they only limit requests that would've failed anyway under previous code.)

- Closes #32: the editor now validates the event and shows immediate feedback for errors
  - Also validates some things not properly validated by the TypeBox schema or even backend
  - maxLengths added to various fields, so that the user can't even input invalid data
  image
- Closes #64: admins can now preview events from the editor, including the signup form
  image
- Signup start/end time and signup publicity are now part of "basic details", to make the experience less confusing
- Added a price field to the editor (the schema already supported this)
- Signup closure date is now shown in event list & event details when the signup is closed
- Quota size fields now have placeholders to indicate what empty means
- Quota name is no longer shown in signup lists when there's only one
- TypeBox schemas updated to use Composite, and they now validate more invariants (instead of failing at database level)
- Various bug fixes to the editor and other UI

### CODE REVIEW

File: packages/ilmomasiina-models/src/schema/event/attributes.ts [hunk 1]
```diff
@@ -4,12 +4,18 @@ import { Nullable } from "../utils";
 
 export const eventSlug = Type.String({
   description: "Event slug, used for accessing the event by URL.",
+  minLength: 1,
```
Comment (lanttu1243) [inline packages/ilmomasiina-models/src/schema/event/attributes.ts:7]: Should the slug be always longer than one character maybe say min. 4

File: packages/ilmomasiina-models/src/schema/event/attributes.ts [hunk 2]
```diff
@@ -4,12 +4,18 @@ import { Nullable } from "../utils";
...
+  maxLength: 255,
```
Comment (lanttu1243) [inline packages/ilmomasiina-models/src/schema/event/attributes.ts:8]: Same for this, is a 255 character slug too long cause then you have the hash on top for signup edits?

File: packages/ilmomasiina-models/src/schema/event/attributes.ts (hunk 1 above)
Comment (PurkkaKoodari) [reply packages/ilmomasiina-models/src/schema/event/attributes.ts:7]: Might be desirable, but then users might be unable to save events they've created previously without changing the URL. So we might need to trust the (admin) users here. (These are the limits previously only enforced by the Sequelize model.)

File: packages/ilmomasiina-models/src/schema/event/attributes.ts (hunk 2 above)
Comment (PurkkaKoodari) [reply packages/ilmomasiina-models/src/schema/event/attributes.ts:8]: The event slug is not present in signup edits

**Review by lanttu1243 — APPROVED** [review]

**Pull request merged**